### PR TITLE
Use crd to manage tidbcluster in stability

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

These request let tidbcluster deployed by CR and fix some stability bugs.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -142,45 +144,45 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
...
+		
 		for _, clusterCfg := range clusterCfgs {
 			localCfg := clusterCfg
 			ginkgo.It(fmt.Sprintf("[TiDB Version: %s] %s", localCfg.Version, localCfg.Name), func() {
 				cluster := newTidbCluster(ns, localCfg.Name, localCfg.Version)
 				cluster.Spec.EnablePVReclaim = pointer.BoolPtr(true)
 				// support reclaim pv when scale in tikv or pd component
 
-				tests.CreateTidbClusterOrDie(cli, cluster)
+				crdUtil.CreateTidbClusterOrDie(cluster)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go:155]: in the future, can we remove these wrappers? in ginkgo e2e, we don't need to call slack.

Comment (Yisaer) [reply tests/e2e/tidbcluster/tidbcluster.go:155]: I think we can add a switch to control it.

Comment (cofyc) [reply tests/e2e/tidbcluster/tidbcluster.go:155]: we can remove thin wrappers such as`CreateTidbClusterOrDie` which just call methods on kubernetes client

File: tests/cmd/stability/main.go [hunk 2]
```diff
@@ -155,58 +169,49 @@ func run() {
...
+		//	crdUil.CheckDisasterToleranceOrDie(tc)
+		//}
+		//clusterZero := clusters[0].Clustrer
+		//clusterZero = crdUil.GetTidbClusterOrDie(clusterZero.Name, clusterZero.Namespace)
+		//clusterZero = fixture.AddPumpForTidbCluster(clusterZero)
+		//crdUil.UpdateTidbClusterOrDie(clusterZero)
+		//crdUil.WaitTidbClusterReadyOrDie(clusterZero, 30*time.Minute)
+		//oa.BackupAndRestoreToMultipleClustersOrDie(clusters[0], backupTargets)
+		//klog.Infof("clusters backup and restore checked")
+		//slack.NotifyAndCompletedf("stability test: clusters backup and restore checked")
```
Comment (cofyc) [inline tests/cmd/stability/main.go:214]: This case does not involve destructive actions. I think we can delete them and migrate them to ginkgo e2e suite (if no corresponding case exists).

File: tests/cmd/stability/main.go [hunk 3]
```diff
@@ -311,7 +319,7 @@ func run() {
 				fta.StartKubeAPIServerOrDie(vNode.IP)
 			}
 		}
-		klog.Infof("clusters stop all kube-apiserver and restart")
+		//klog.Infof("clusters stop all kube-apiserver and restart")
```
Comment (cofyc) [inline tests/cmd/stability/main.go:322]: revert

File: tests/crd_test_util.go [hunk 4]
```diff
@@ -122,3 +156,463 @@ func checkPodsDisasterTolerance(allPods []corev1.Pod) error {
...
+	err := ctu.cli.PingcapV1alpha1().TidbClusters(tc.Namespace).Delete(tc.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return
+		}
+		slack.NotifyAndPanic(err)
+	}
+}
+
+func (ctu *CrdTestUtil) WaitTidbClusterReadyOrDie(tc *v1alpha1.TidbCluster, timeout time.Duration) {
```
Comment (cofyc) [inline tests/crd_test_util.go:170]: why not use the similary function `operatorActions.WaitForTidbClusterReady` which takes a TidbCluster CR too?

Comment (Yisaer) [reply tests/crd_test_util.go:170]: I plan we will finally discard the function which operate the tidbcluster in `operatorActions`

File: tests/cmd/stability/main.go (hunk 2 above)
Comment (Yisaer) [reply tests/cmd/stability/main.go]: updated

File: tests/crd_test_util.go (hunk 4 above)
Comment (cofyc) [reply tests/crd_test_util.go:170]: ok

File: tests/cmd/stability/main.go [hunk 5]
```diff
@@ -124,29 +127,40 @@ func run() {
 	oa.CleanOperatorOrDie(ocfg)
 	oa.DeployOperatorOrDie(ocfg)
 
+	crdUil := tests.NewCrdTestUtil(cli, kubeCli, asCli, false)
```
Comment (cofyc) [inline tests/cmd/stability/main.go:130]: s/crdUil/crdUilt`

I guess we should use `Util` here, which is short for `utility`

**Pull request merged**